### DISCUSSION TITLE
[FEATURE REQUEST] salt-cloud AWS/EC2 Spot Request InstanceInterruptionBehavior "stop"?

### DISCUSSION THREAD

**Discussion by choffman** [saltstack/salt#67509] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
When making an AWS Spot request, Instance Interruption Behavior is always set to "terminate".  This causes a problem if an administrator accidentally issues a shutdown command.  The machine is thus destroyed. 

**Describe the solution you'd like**
I would like the ability to define Instance Interruption Behavior on my AWS Spot requests.  The default is "terminate", but I would like to be able to set it to "stop".  

**Describe alternatives you've considered**
The alternative is shutting the machine down from the EC2 Web console.  I would prefer to have the ability to define this at a machine level.

**Additional context**

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct] (https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html)
  - [Join our Community Slack](https://saltstackcommunity.herokuapp.com/)
  - [IRC on Freenode](https://webchat.freenode.net/#salt)
  - [SaltStack YouTube channel](https://www.youtube.com/user/SaltStack)
  - [SaltStackInc Twitch channel](https://www.twitch.tv/saltstackinc)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at core@saltstack.com or reach out directly to the Community Manager, Cassandra Faris via Slack. We’re glad you’ve joined our community and look forward to doing awesome things with you!
